includes/API/ApiLabkiUpdate.php

```diff
@@ -11,165 +11,336 @@
-                if ( $contentRepoUrl === '' || $packName === '' ) {
-                    $this->dieWithError( 'apierror-missing-param' );
-                }
-                $repoId = $repoReg->ensureRepo( $contentRepoUrl );
-                $packIdObj = $packReg->getPackIdByName( $repoId, $packName, null );
-                if ( $packIdObj === null ) {
-                    $this->dieWithError( [ 'apierror-pack-not-found', $packName ], 'pack_not_found' );
-                }
-                $packId = $packIdObj instanceof PackId ? $packIdObj->toInt() : (int)$packIdObj;
-                if ( $newVersion !== null && $newVersion !== '' ) {
-                    $packReg->updatePack( $packId, [ 'version' => $newVersion ] );
-                } else {
-                    $packReg->updatePack( $packId, [] ); // touch updated_at
-                }
-                // Replace pages
-                $pageReg->removePagesByPack( $packId );
-                $added = 0;
-                foreach ( $pages as $p ) {
-                    if ( !is_array( $p ) ) { continue; }
-                    $name = (string)($p['name'] ?? '');
-                    $finalTitle = (string)($p['finalTitle'] ?? '');
-                    $ns = (int)($p['pageNamespace'] ?? 0);
-                    $wikiPageId = (int)($p['wikiPageId'] ?? 0);
-                    if ( $name === '' || $finalTitle === '' ) { continue; }
-                    $pageReg->recordInstalledPage( $packId, $name, $finalTitle, $ns, $wikiPageId );
-                    $added++;
-                }
-                $result['success'] = true;
-                $result['packId'] = $packId;
-                $result['pagesAdded'] = $added;
-                break;
+            $pageReg->removePagesByPack( $packId );
+            foreach ( $createdPages as $p ) {
+                $pageReg->recordInstalledPage(
+                    $packId,
+                    $p['name'],
+                    $p['finalTitle'],
+                    $p['pageNamespace'],
+                    $p['wikiPageId']
+                );
+            }
 
-            default:
-                $result['error'] = 'unknown_action';
+            $imported[] = [
+                'pack' => $packName,
+                'pages' => count( $createdPages )
+            ];
         }
 
-        $this->getResult()->addValue( null, $this->getModuleName(), $result );
+        return [
+            'success' => true,
+            'action' => 'installPack',
+            'imported' => $imported,
+            'rewriteEntries' => count( $rewriteMap ),
+            '_meta' => [ 'schemaVersion' => 1 ]
+        ];
     }
 
+    /* -------------------------------------------------------------------------
+     * UPDATE PACK
+     * ------------------------------------------------------------------------- */
+    private function handleUpdatePack(): array {
```

Comment:
not seeing where the contents of the pages get updated here, it looks like the `install` and `remove` methods both actually mutate the wiki, but this looks like it just modifies the registry?

Reply:
I only have`handleImportPack()` up and running right now. Next will be doing pack updates and pack removals.

So for now Update and Remove don't do anything they are supposed to do. Just like `handleImportPack()`, these other methods should both modify MW pages and, when successful, update DB tables.